### ISSUE TITLE
FALSE NEGATIVE: `DM_USELESS_THREAD` misses a `Thread` subclass that does not override `run()`

### ISSUE THREAD

**Issue by lynae99** [spotbugs/spotbugs#4306]
Hi, I found a false negative in SpotBugs 4.10.3 when a class extends `Thread` without overriding `run()` or passing a `Runnable` target.

**Affected tool**

`SpotBugs 4.10.3`

**Affected checker**

`SpotBugs DM_USELESS_THREAD`

**Minimal reproducer**

```java
class SpotBugsSonarQubeDMUselessThread {
    public static class Subject extends Thread {
    }

    void test() {
        new Subject().start();
    }
}
```

**Reproduction command**

```bash
spotbugs -version
javac -d classes spotbugs-sonarqube-dm-useless-thread-fn-thread-subclass.java
spotbugs -textui -effort:max -low -visitors DumbMethods classes
```

**Current behavior**

SpotBugs produces no `DM_USELESS_THREAD` diagnostic.

**Expected behavior**

SpotBugs should report the `Subject` thread creation because its inherited default `run()` method performs no action and no `Runnable` target is provided.
